include/mem.h

```diff
@@ -89,9 +89,9 @@ static inline uint32_t var_read(uint32_t *var)
 /* The Following six functions are slower but they recognize the paged memory
  * system */
 
-uint8_t mem_readb(PhysPt pt);
-uint16_t mem_readw(PhysPt pt);
-uint32_t mem_readd(PhysPt pt);
+uint8_t mem_readb(PhysPt pt, bool from_debugger = false);
```

Comment:
This code adds additional argument to functions which are performance critical. How about

```C++
#if C_DEBUG && C_HEAVY_DEBUG
uint8_t mem_readb(PhysPt pt, bool from_debugger = false);
(…)
#else
uint8_t mem_readb(PhysPt pt);
(…)
#endif
```

Reply:
the real question is: why there are not header-only but splitted into header/cpp - while the inner called function is also a header-only inline ???

Reply:
looks better - so absolutely no impact using C_HEAVY_DEBUG
question is: is there a real need to encapsulate the paging.h stuff - else these 6 read/write routines could be also inlined - or do we already use LTO or stuff like that